core/src/avm2/globals/flash/events/textevent.rs

```diff
@@ -0,0 +1,103 @@
+use crate::avm2::activation::Activation;
+use crate::avm2::class::{Class, ClassAttributes};
+use crate::avm2::events::EventData;
+use crate::avm2::method::{Method, NativeMethodImpl};
+use crate::avm2::names::{Namespace, QName};
+use crate::avm2::object::{Object, TObject};
+use crate::avm2::value::Value;
+use crate::avm2::Error;
+use gc_arena::{GcCell, MutationContext};
+
+/// Implements `flash.events.TextEvent`'s instance constructor.
+fn instance_init<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        activation.super_init(this, args)?; // Event uses these
+        if let Some(mut evt) = this.as_event_mut(activation.context.gc_context) {
+            let text = args
+                .get(3)
+                .cloned()
+                .unwrap_or(Value::Undefined)
+                .coerce_to_string(activation)?;
+            evt.set_event_data(EventData::Text { text });
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+/// Implements `flash.events.TextEvent`'s class constructor.
+fn class_init<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+/// Implements `text`'s setter.
+fn set_text<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Some(mut evt) = this.as_event_mut(activation.context.gc_context) {
+            let text = args
+                .get(0)
+                .cloned()
+                .unwrap_or(Value::Undefined)
+                .coerce_to_string(activation)?;
+            evt.set_event_data(EventData::Text { text });
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `text`'s getter.
+fn text<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Some(evt) = this.as_event() {
+            if let EventData::Text { text, .. } = evt.event_data() {
```

Comment:
Subclasses of `TextEvent` will also have a `text` property; this currently will return `undefined` for all subclasses that have a different native `Error` type.

Reply:
This seems complicated to fix. Is the way to fix it without removing the enum to add the same method for each inherited class? And also adding IOErrorEvent and SecurityErrorEvent to the EventData enum with the exact same attributes as ErrorEvent (which, as you noted,  needs TextEvent's attributes added to it)?